### PR TITLE
[BUGFIX] Implement Focus Out Event Handling for Input Controls

### LINKED ISSUE #2023: [🐛 BUG] [UX] input, number, date controls don't validate on focus out event
### What went wrong? 🤔

The input control (as well as number, data, ...) does not trap the Focus Out event that should trigger the equivalent of pressing the Enter key

### Expected Behavior

See description.

All `input`, `number`, `date` should be impacted.
`login` probably...

image

In this case, I don't want to be pressing Enter before "Share".

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by nagarajpandith**
This PR adds an onBlur event handler to the input control in Taipy GUI, ensuring that input validation is triggered when the user focuses out of the input field. Fixes #2023 @FabienLelaquais @jrobinAV

### Changes:
- Added `handleBlur` function to dispatch the update action when the input field loses focus.
- Ensured that all input types (text, number, date) are affected by this change.

### Acceptance Criteria:
- Input controls should now validate on focus out, equivalent to pressing the Enter key.
- All new code is unit tested with a code coverage of at least 90%.
Screenshot 2024-10-12 at 9 37 33 PM

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Input.tsx [hunk 1]
```diff
@@ -197,6 +197,11 @@ const Input = (props: TaipyInputProps) => {
         ]
     );
 
+    const handleBlur = (e : React.FocusEvent<HTMLInputElement>) => {
+        const val = e.target.value;
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Input.tsx:201]: you need to stop the timer if any

Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Input.tsx:201]: use currentTarget

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Don't see no tests

**Review by UTK-a-RSH — APPROVED** [review]

**Comment by FredLL-Avaiga** [comment]
@UTK-a-RSH Thanks for *not* approving PR :-)

**Comment by FredLL-Avaiga** [comment]
Hi @nagarajpandith, are you still working on this ?
If you are, please, answer the questions
If you are not, please close this PR

**Pull request closed without merging**